Content/RootScript/AngelEngine/Utility/Util.as

```diff
@@ -0,0 +1,329 @@
+// -------------------------------------------------------------------- //
+//									//
+//			SCP : Containment Breach			//
+//									//
+// -------------------------------------------------------------------- //
+// Script: RootScript/Utility/Util.as					//
+// Source: https://github.com/juanjp600/scpcb/Content/			//
+// Script File Author(s): Pyro-Fire					//
+// Purpose:								//
+//	- General purpose utilities					//
+//									//
+//									//
+// -------------------------------------------------------------------- \\
+// Documentation
+//
+//	SECTION 1. AngelMath
+//		- GUI::Align : GUI::Align enumerator
+//		- Util::FloatInterpolator : Number smoother
+//
+//	SECTION 2. AngelString
+//		- Icon <class> : Abstractable Icon@ generator definition.
+//		- Icon::Model : Icon@ which generates a texture from a model.
+//
+//	SECTION 3. AngelUI
+//		- drawSquare : Draws a textured, colored Rectanglef.
+//
+//	SECTION 4. Icon Handler
+//		- Util::Icon <class> : Abstractable Icon@ generator definition.
+//		- Util::Icon::Model : Icon@ which generates a texture from a model.
+//
+//	SECTION 5. Hook
+//		- Hook <namespace> : Create and destroy hooks
+//		- Hook@ : call hooks, and add to hook.
+//		Usage:
+//			Hook::fetch(name).add(@Hook::Function);
+//
+//
+//									//
+// -------------------------------------------------------------------- \\
+// Begin Script
+
+// # Util::Function@, TickFunction and RenderFunction, and noops ----
+namespace Util {
+	shared funcdef void Function();
+	shared void noop() {}
+	shared funcdef void TickFunction(uint32 tick, float interp);
+	shared void noopTick(uint32 tick, float interp) {}
+	shared funcdef void RenderFunction(float interp);
+	shared void noopRender(float interp) {}
+}
+
+// #### SECTION 1. AngelMath ----
+
+// # AngelMath ----
+// Numeracy functions, definitions, libraries etc
+namespace Math {
+	shared float pi2f=Math::PI*2.f;
+}
+namespace Util {
+	shared float fpsFactor(float interp) { return Math::maxFloat(Math::minFloat(interp*70.f,5.f),0.2f); } // Original math
```

Comment:
We're not using FPSFactor anymore. This is not permissible. Remove it.

Reply:
I don't recall this being used anywhere currently because the numbers it gave were crazy, impractical, and not useful.
How did it even work in the original?
It's in milliseconds, right?
There's also an FPSFactor2 that I could not figure out what it did.

Reply:
FPSFactor is a value that is inversely proportional to the framerate, with its value being 1 at 70 FPS. I imagine the reasoning behind this was that at some point the game didn't attempt to account for variable framerate at all, being locked at 60 FPS or something, so FPSfactor was added and then boosted a little bit to make things go faster. That being said, it's a horribly confusing way of going about it, so we prefer deltaTime, which is just the length of a tick in seconds.

Reply:
Thank you for this!
I have added what it is to my notes and removed the snippet.